src/core_plugins/input_control_vis/public/components/editor/index_pattern_select.js

```diff
@@ -1,45 +1,125 @@
+import _ from 'lodash';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import Select from 'react-select';
 
 import {
   EuiFormRow,
+  EuiComboBox,
 } from '@elastic/eui';
 
 export class IndexPatternSelect extends Component {
   constructor(props) {
     super(props);
 
-    this.loadOptions = this.loadOptions.bind(this);
+    this.state = {
+      isLoading: false,
+      options: [],
+      selectedIndexPattern: undefined,
+    };
   }
 
-  loadOptions(input, callback) {
-    this.props.getIndexPatterns(input).then((indexPatternSavedObjects) => {
+  componentWillMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    this.debouncedFetch.cancel();
+  }
+
+  componentDidMount() {
+    this.fetchOptions();
+    this.fetchSelectedIndexPattern(this.props.indexPatternId);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.indexPatternId !== this.props.indexPatternId) {
+      this.fetchSelectedIndexPattern(nextProps.indexPatternId);
+    }
+  }
+
+  fetchSelectedIndexPattern = async (indexPatternId) => {
+    if (!indexPatternId) {
+      this.setState({
+        selectedIndexPattern: undefined
+      });
+      return;
+    }
+
+    const indexPattern = await this.props.getIndexPattern(indexPatternId);
+
+    if (!this._isMounted) {
+      return;
+    }
+
+    // TODO: handle case where index pattern no longer exists
```

Comment:
Did you mean to leave this TODO in the checked in code, or should it actually be done before merging this PR?